packages/web-wallet/src/utils/hathor.ts

```diff
@@ -0,0 +1,118 @@
+ */
+export const formatHTRAmount = (amount: number | bigint, isNft: boolean = false): string => {
+  // prettyValue handles bigint | number | string natively, no conversion needed
+  const decimalPlaces = isNft ? 0 : DECIMAL_PLACES;
+  return prettyValue(amount, decimalPlaces);
+};
+
+/**
+ * Parse HTR amount string to cents (satoshis) using precise BigInt arithmetic.
+ * Avoids floating-point precision loss by using string manipulation and BigInt.
+ *
+ * @example
+ * htrToCents("123.45") // 12345n
+ * htrToCents("10") // 1000n
+ * htrToCents("0.01") // 1n
+ *
+ * @param amount Amount as string (e.g., "10.50")
+ * @returns Amount in cents as BigInt
+ * @throws Error if amount format is invalid
+ */
+export const htrToCents = (amount: string): bigint => {
+  // Validate format - must be digits with optional decimal (max 2 places)
+  const trimmed = amount.trim();
+  if (!/^\d+(\.\d{1,2})?$/.test(trimmed)) {
+    throw new Error('Invalid amount format. Use up to 2 decimal places.');
+  }
+
+  // Split into integer and decimal parts
+  const [integerPart, decimalPart = ''] = trimmed.split('.');
+
+  // Pad decimal to 2 places: "5" -> "50", "5" -> "50", "" -> "00"
+  const paddedDecimal = decimalPart.padEnd(DECIMAL_PLACES, '0');
+
+  // Construct cents value: combine integer and decimal parts
+  // Example: "123.45" -> "123" + "45" -> 12345n
+  const centsString = integerPart + paddedDecimal;
+  const result = BigInt(centsString);
+
+  // Validate positive after BigInt conversion (avoids parseFloat precision issues)
+  if (result <= 0n) {
+    throw new Error('Amount must be greater than 0');
+  }
+
+  return result;
+};
+
+/**
+ * Convert cents (satoshis) to HTR decimal string for form inputs.
+ * Returns plain numeric string without formatting (no thousand separators).
+ *
+ * @example
+ * centsToHTR(12345n) // "123.45"
+ * centsToHTR(1000000n) // "10000.00"
+ * centsToHTR(1n) // "0.01"
+ *
+ * @param cents Amount in cents (can be number or BigInt)
+ * @returns Amount as string with decimal point
+ */
+export const centsToHTR = (cents: number | bigint): string => {
+  const amount = toBigInt(cents);
+
+  // Perform BigInt division and modulo to split integer and decimal parts
+  const integerPart = amount / DECIMAL_MULTIPLIER;
+  // For negative amounts, modulo returns negative, so take absolute value
+  const decimalPart = amount < 0n ? -(amount % DECIMAL_MULTIPLIER) : (amount % DECIMAL_MULTIPLIER);
+
+  // Format decimal part with leading zeros if needed
+  // Example: 5n -> "05", 50n -> "50"
+  const decimalString = decimalPart.toString().padStart(DECIMAL_PLACES, '0');
+
+  return `${integerPart}.${decimalString}`;
+};
+
+/**
+ * Truncate address for display
+ * @param address Full address
+ * @param prefixLength Length of prefix to show
+ * @param suffixLength Length of suffix to show
+ * @returns Truncated address
+ */
+export const truncateAddress = (address: string, prefixLength = 7, suffixLength = 7): string => {
```

Comment:
You can use this to truncate also uids. Maybe we could rename to `truncateString` and use in other places